tests/functional/test_templates.py

```diff
@@ -0,0 +1,54 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from jinja2 import Environment, FileSystemLoader
+
+
+def test_templates_for_empty_titles():
+    """
+        Test if all HTML templates have defined the title block. See
+        https://github.com/pypa/warehouse/issues/784
+    """
+    dir_name = os.path.dirname(__file__)
+    dir_name = os.path.join(dir_name, '..', '..', 'warehouse', 'templates')
```

Comment:
This should probably ``import warehouse`` and then do ``os.path.join(os.path.dirname(warehouse.__file__), "templates")`` to make this more resilient against the test code getting shuffled around.